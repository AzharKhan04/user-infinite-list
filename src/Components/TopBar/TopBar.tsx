import React from "react";
import AuthService from "../../Services/auth.service";
import { ExitToAppOutlined } from "@mui/icons-material";
import { useNavigate } from "react-router";


const TopBar: React.FunctionComponent<any> = () => {

    const navigate = useNavigate()

    const performLogout = () => {

        let authService : AuthService = new AuthService();
        authService.logout();
        navigate('/login')
    }

    return (
        <div className="ui-navigator">
        <div className="row">
          <div
            style={{ paddingTop: "18px", paddingLeft: "40px" }}
            className="col-6 text-left"
          >
            <h5 style={{ fontSize: "20px", color: "royalblue" }}>
              <strong>{"Users"}</strong>
            </h5>
          </div>
          <div
            onClick={performLogout}
            style={{
              paddingTop: "18px",
              paddingLeft: "40px",
              cursor: "pointer",
            }}
            className="col-6 text-right"
          >
            <h5
              style={{
                textAlign: "center",
                fontSize: "20px",
                color: "royalblue",
              }}
            >
              <ExitToAppOutlined />
              {" Logout"}
            </h5>
          </div>
        </div>
      </div>
    )

}

export default React.memo(TopBar);