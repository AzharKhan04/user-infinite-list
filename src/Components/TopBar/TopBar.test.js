import renderer from 'react-test-renderer';
import TopBar from './TopBar';
import {Router} from 'react-router-dom'
import { render, screen } from '@testing-library/react';

describe('TopBar Component', () => {
  

it('TopBar component renders correctly', () => {

  const location = {  };
  const tree = renderer.create(<Router location={location}><TopBar/></Router>).toJSON();
  expect(tree).toMatchSnapshot();


});


test('Topbar component has text users ',  () => {

    const location = { pathname: '' };
    render(<Router location={location}><TopBar/></Router>)
  
    expect(screen.getByText("Users")).toBeInTheDocument();
     
  })
})