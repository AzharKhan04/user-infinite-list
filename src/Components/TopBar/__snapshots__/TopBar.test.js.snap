// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Component TopBar component renders correctly 1`] = `
<div
  className="ui-navigator"
>
  <div
    className="row"
  >
    <div
      className="col-6 text-left"
      style={
        Object {
          "paddingLeft": "40px",
          "paddingTop": "18px",
        }
      }
    >
      <h5
        style={
          Object {
            "color": "royalblue",
            "fontSize": "20px",
          }
        }
      >
        <strong>
          Users
        </strong>
      </h5>
    </div>
    <div
      className="col-6 text-right"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "paddingLeft": "40px",
          "paddingTop": "18px",
        }
      }
    >
      <h5
        style={
          Object {
            "color": "royalblue",
            "fontSize": "20px",
            "textAlign": "center",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExitToAppOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.09 15.59 11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
         Logout
      </h5>
    </div>
  </div>
</div>
`;
